### PR TITLE
Fix MobileILStrip handling of unmanaged dlls

### PR DESCRIPTION
**Pull request by jkurdek**
Fixes https://github.com/dotnet/runtime/issues/101967
Follow up to the runtime change https://github.com/dotnet/runtime/pull/106267

### CODE REVIEW

**Comment by rolfbjarne** [comment]
> @rolfbjarne it seems to me after inspection of binlogs that ILstrip was not run for any of the test. I can see the `_RunAotCompiler` property being false.

The bots were probably all x64 bots.

I'll try locally and see what I get.

**Comment by rolfbjarne** [comment]
It seems the ResolveAssemblyReferences task just removes the reference to NativeLibrary.dll:

> /Users/rolf/work/maccore/net9.0/xamarin-macios/builds/downloads/dotnet-sdk-9.0.100-rc.2.24420.1/sdk/9.0.100-rc.2.24420.1/Microsoft.Common.CurrentVersion.targets(2413,5): warning MSB3246: Resolved file has a bad image, no metadata, or is otherwise inaccessible. Invalid number of sections declared in PE header. Invalid number of sections declared in PE header. [/Users/rolf/work/maccore/net9.0/xamarin-macios/tests/xharness/bin/Debug/tmp-test-dir/monotouch-test1066/monotouch-test.csproj]

So it never reaches the ILStrip target.

Binlog: 
[build-iOS_Unified64-20240902_191140.binlog.zip](https://github.com/user-attachments/files/16840545/build-iOS_Unified64-20240902_191140.binlog.zip)

**Comment by jkurdek** [comment]
Thanks @rolfbjarne, the dll was referenced incorrectly. I did verify locally that without changing the MobileILStrip task we get a failure, and that the proposed fix works.

File: tests/monotouch-test/dotnet/shared.csproj [hunk 1]
```diff
@@ -272,4 +272,8 @@
       <DefineConstants Condition="'$(RuntimeIdentifier)' == 'tvossimulator-arm64'">$(DefineConstants);AOT</DefineConstants>
     </PropertyGroup>
   </Target>
+  <PropertyGroup>
+    <ProvisioningType>manual</ProvisioningType>
+    <CodesignKey>Apple Development: jkurdek@gmail.com (896D8S3LKR)</CodesignKey>
+</PropertyGroup>
```
Comment (rolfbjarne) [inline tests/monotouch-test/dotnet/shared.csproj:278]: This won't work on the bots :)

**Review by rolfbjarne — CHANGES_REQUESTED** [review]

File: tests/monotouch-test/dotnet/shared.csproj (hunk 1 above)
Comment (jkurdek) [reply tests/monotouch-test/dotnet/shared.csproj]: Missed this 🥶, thanks!

**Comment by rolfbjarne** [comment]
There's a test failure in the .NET tests:

```
Xamarin.Tests.PostBuildTest.AssemblyStripping("MySimpleAppWithSatelliteReference",iOS,"ios-arm64",True): Application Assembly
Expected: file or directory exists
But was: "/Users/builder/azdo/_work/4/s/xamarin-macios/tests/dotnet/MySimpleAppWithSatelliteReference/iOS/bin/Debug/net9.0-ios/ios-arm64/MySimpleAppWithSatelliteReference.app/MySimpleAppWithSatelliteReference.dll"
```

I can reproduce by executing:

```
$ make run-tests -C tests/dotnet/UnitTests TEST_FILTER="--filter FullyQualifiedName~Xamarin.Tests.PostBuildTest.AssemblyStripping"
```

File: eng/Versions.props [hunk 2]
```diff
@@ -9,7 +9,7 @@
     <MicrosoftDotNetSharedFrameworkSdkVersion>8.0.0-beta.24413.2</MicrosoftDotNetSharedFrameworkSdkVersion>
     <MicrosoftNETCoreAppRefPackageVersion>9.0.0-rc.2.24462.10</MicrosoftNETCoreAppRefPackageVersion>
     <MicrosoftNETWorkloadEmscriptenCurrentManifest80100TransportVersion>8.0.0-rtm.23511.3</MicrosoftNETWorkloadEmscriptenCurrentManifest80100TransportVersion>
-    <MicrosoftNETRuntimeMonoTargetsSdkPackageVersion>9.0.0-alpha.1.23556.4</MicrosoftNETRuntimeMonoTargetsSdkPackageVersion>
+    <MicrosoftNETRuntimeMonoTargetsSdkPackageVersion>9.0.0-rc.2.24462.10</MicrosoftNETRuntimeMonoTargetsSdkPackageVersion>
```
Comment (jkurdek) [inline eng/Versions.props:12]: I think should not be doing this manually, @rolfbjarne what is the correct way to bump this version?

Comment (rolfbjarne) [reply eng/Versions.props:12]: Yes, this is fine, we just don't have a maestro subscription for this dependency (it rarely needs updating).

**Review by rolfbjarne — APPROVED** [review]

**Review by emaf — APPROVED** [review]

**Review by emaf — APPROVED** [review]

**Pull request merged**